pkg/templates/templates.go

```diff
@@ -326,6 +326,10 @@ func (template *Template) UnmarshalYAML(unmarshal func(interface{}) error) error
 	}
 	*template = Template(*alias)
 
+	if !ReTemplateID.MatchString(template.ID) {
```

Comment:
We should consider adding checks for other mandatory fields as well. See parser_validate.go:validateTemplateMandatoryFields